awx/main/scheduler/kubernetes.py

```diff
@@ -0,0 +1,74 @@
+import time
+
+from django.conf import settings
+from kubernetes.client import Configuration as kubeconfig
+from kubernetes.client import CoreV1Api as v1_kube_api
+from kubernetes.client import ApiClient as kube_api_client
+
+from awx.main.utils.common import parse_yaml_or_json
+
+
+class PodManager(object):
+    credential = None
+    kube_api = None
+
+    def __init__(self, task):
+        self.task = task
+        self.credential = task.instance_group.credential
+        self.kube_api = self._kube_api_connection(self.credential)
+        self.pod_name = f"job-{task.id}"
+        self.pod_definition = self._pod_definition()
+
+    def deploy_pod(self):
+        if not self.credential.kubernetes:
+            raise RuntimeError
+
+        self.kube_api.create_namespaced_pod(body=self.pod_definition,
+                                            namespace='awx')
+
+        while True:
```

Comment:
Is there some way we can make it so this loop is _guaranteed_ to end eventually? I see that we're checking against `pod.status`, but maybe some upper limit or retry count would be good here as a fallback?

Reply:
I ignored this at first with the intention of circling back around and now I am wondering if we need this kind of logic here at all because a) the job itself will hit a timeout if it can't start after a certain amount of time. See `AWX_ISOLATED_LAUNCH_TIMEOUT`. and b) Kubernetes itself will not stay in Pending forever - it will either error out or start. But I hear you. Just not sure if we need it given what I stated. What do you think?

Reply:
That makes sense. It sounds like `AWX_ISOLATED_LAUNCH_TIMEOUT` should have this covered - sounds good to me.